### PR TITLE
Refactor pipelines

### PR DESCRIPTION
**Pull request by balanza**
#### List of Changes

* remove reference to legacy versions of azure templates
* remove local template for project setup 
* fix healthcheck endpoint

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .devops/code-review-pipelines.yml [hunk 1]
```diff
@@ -19,6 +15,14 @@ trigger: none
 pool:
   vmImage: 'ubuntu-latest'
 
+resources:
+  repositories:
+    - repository: pagopaCommons
+      type: github
+      name: pagopa/azure-pipeline-templates
+      ref: refs/tags/v14
+      endpoint: 'pagopa'
```
Comment (pasqualedevita) [inline .devops/code-review-pipelines.yml:24]:
```suggestion
      endpoint: 'io-azure-devops-github-ro'
```

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -28,16 +24,10 @@ pool:
...
-      name: pagopa/azure-pipeline-templates
-      ref: refs/tags/v4
-      endpoint: 'pagopa'
-
     - repository: pagopaCommons
       type: github
       name: pagopa/azure-pipeline-templates
-      ref: refs/tags/v13
+      ref: refs/tags/v14
       endpoint: 'pagopa'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:31]:
```suggestion
      endpoint: 'io-azure-devops-github-ro'
```

File: .devops/yarn-lock-upgrade.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ resources:
     - repository: pagopaCommons
       type: github
       name: pagopa/azure-pipeline-templates
-      ref: refs/tags/v12
+      ref: refs/tags/v14
       endpoint: 'pagopa'
```
Comment (pasqualedevita) [inline .devops/yarn-lock-upgrade.yml:24]:
```suggestion
      endpoint: 'io-azure-devops-github-ro'
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Comment by pasqualedevita** [comment]
change also here `api/v1/info`
https://github.com/pagopa/io-functions-template/blob/update-pipelines/Info/index.ts#L13

I don't known if we need to add also
`"route": "v1/info",`
https://github.com/pagopa/io-functions-pushnotifications/blob/master/Info/function.json#L8

**Comment by pasqualedevita** [comment]
fix also danger issue

**Comment by gquadrati** [comment]
> change also here `api/v1/info`
> https://github.com/pagopa/io-functions-template/blob/update-pipelines/Info/index.ts#L13
> 
> I don't known if we need to add also
> `"route": "v1/info",` like
> https://github.com/pagopa/io-functions-pushnotifications/blob/master/Info/function.json#L8

Done in [8d8f5ca](https://github.com/pagopa/io-functions-template/pull/21/commits/8d8f5ca6f59941a11eebea0445956990a5d87524)

**Review by pasqualedevita — COMMENTED** [review]
@balanza fix danger issue is out of scope?

**Comment by gquadrati** [comment]
> @balanza fix danger issue is out of scope?

@pasqualedevita @balanza fixed in #23

**Review by gquadrati — APPROVED** [review]
LGTM

**Review by pasqualedevita — APPROVED** [review]
LGTM

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]
If the express route is `api/v1/info` and the functions.json route is `v1/info`, we need to add `routePrefix=api` inside host.json file. Otherwise the route inside functions.json must be `api/v1/info` accordingly with the express route.

**Comment by gquadrati** [comment]
> If the express route is `api/v1/info` and the functions.json route is `v1/info`, we need to add `routePrefix=api` inside host.json file. Otherwise the route inside functions.json must be `api/v1/info` accordingly with the express route.

Done in [83ab661](https://github.com/pagopa/io-functions-template/pull/21/commits/83ab6612934972d449b5c1a72387435de6a1a3c6)

File: Info/function.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
       "type": "httpTrigger",
       "direction": "in",
       "name": "req",
-      "route": "template/info",
+      "route": "/api/v1/info",
```
Comment (BurnedMarshal) [inline Info/function.json:8]: Be careful that the initial `/` doesn't work on production.
```suggestion
      "route": "api/v1/info",
```

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Pull request merged**